core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -237,6 +250,24 @@ public boolean sendSync(Transmission transmission) {
         return true;
     }
 
+    private static void handleTemporaryExceptions(String message, Exception ex) {
+        //Handle first failure
+        if(!firstFailure.getAndSet(true)) {
+            logger.error(message+"\n"+
+                    "Total number of successful telemetry requests so far:"+successCounter.get()+"\n"+
+                    "Future failures will be aggregated and logged once every 5 minutes\n",
+                    ex
+            );
+        //Log the first failure every 5 minutes.
+        } else if(failureCounter.getAndIncrement() == 0) {
+            logger.error(message+"\n"+
+                            "Total number of failed telemetry requests in the last 5 minutes:"+previousFailureCounter.get()+"\n"+
+                            "Total number of successful telemetry requests in the last 5 minutes:"+previousFailureCounter.get()+"\n"+
+                    ex
+            );
+        }
```

Comment:
if we can log directly in the thread that runs every 5-min, then possibly this can simplify:

```suggestion
        } else {
            failureCounter.getAndIncrement();
        }
```

Reply:
good catch:)